apps/main-app/src/containers/allelePage/AlleleToDiseaseTable.js

```diff
@@ -4,21 +4,28 @@ import {
   EvidenceCodesCellCuration,
   ReferencesCellCuration,
 } from '../../components/dataTable';
-import AlleleSource from '../../components/dataTable/AlleleSource';
 import AnnotatedEntitiesPopupCuration from '../../components/dataTable/AnnotatedEntitiesPopupCuration';
 import AssociationType from '../../components/AssociationType';
 import DiseaseLinkCuration from '../../components/disease/DiseaseLinkCuration';
 import DiseaseQualifiersColumn from '../../components/dataTable/DiseaseQualifiersColumn';
-import { getDistinctFieldValue } from '../../components/dataTable/utils';
+import { 
+  buildProvidersWithUrl,
+  getDistinctFieldValue } from '../../components/dataTable/utils';
+import ProvidersCellCuration from '../../components/dataTable/ProvidersCellCuration';
 import useDataTableQuery from '../../hooks/useDataTableQuery';
 
 const AlleleToDiseaseTable = ({alleleId}) => {
   const {
     resolvedData,
-    data,
+    data: results,
     ...tableProps
   } = useDataTableQuery(`/api/allele/${alleleId}/diseases`);
 
+  const tableData = results.map(annotation => ({
+    providers: buildProvidersWithUrl(annotation.primaryAnnotations),
+    ...annotation,
+  }));
+
```

Comment:
This data should not be manipulated here... just create a component for the table to use.